lib/Launcher.js

```diff
@@ -278,33 +278,33 @@ class Launcher {
   }
 
   /**
-   * @param {!(Launcher.BrowserOptions & {browserWSEndpoint?: string, browserUrl?: string, transport?: !Puppeteer.ConnectionTransport})} options
+   * @param {!(Launcher.BrowserOptions & {browserWSEndpoint?: string, browserURL?: string, transport?: !Puppeteer.ConnectionTransport})} options
    * @return {!Promise<!Browser>}
    */
   async connect(options) {
     const {
       browserWSEndpoint,
-      browserUrl,
+      browserURL,
       ignoreHTTPSErrors = false,
       defaultViewport = {width: 800, height: 600},
       transport,
       slowMo = 0,
     } = options;
 
-    let connectionUrl;
-    let connectionTransport;
+    assert(!!browserWSEndpoint + !!browserURL + !!transport === 1, 'Exactly one of browserWSEndpoint, browserURL or transport must be passed to puppeteer.connect');
 
-    if (browserWSEndpoint)
-      connectionUrl = browserWSEndpoint;
-    else if (!browserWSEndpoint && browserUrl)
-      connectionUrl = await getWSEndpoint(browserUrl);
-
-    if (transport)
-      connectionTransport = transport;
-    else
-      connectionTransport = await WebSocketTransport.create(connectionUrl);
+    let connection = null;
+    if (transport) {
+      connection = new Connection('', transport, slowMo);
```

Comment:
Null instead of empty string?

Reply:
Must be a string per our docs and everything - I'd keep it as-is